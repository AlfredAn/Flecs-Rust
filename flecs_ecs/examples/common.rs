#![allow(dead_code)]

pub use flecs_ecs::{core::*, macros::Component};

#[derive(Debug, Default, Clone, Component)]
pub struct Position {
    pub x: f32,
    pub y: f32,
}

#[derive(Default, Debug, Clone, Component)]
pub struct Velocity {
    pub x: f32,
    pub y: f32,
}

#[derive(Default, Clone, Component)]
pub struct Mass {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct Eats;

#[derive(Default, Clone, Component)]
pub struct Apples;

#[derive(Default, Clone, Component)]
pub struct Pears;

#[derive(Default, Clone, Component)]
pub struct Walking;

#[derive(Default, Clone, Component)]
pub struct Tag;

#[derive(Default, Clone, Component)]
pub struct Human;

#[derive(Default, Clone, Component)]
pub struct Attack {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct Defence {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct Damage {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct FreightCapacity {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct ImpulseSpeed {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct HasFlt;

#[derive(Default, Clone, Component)]
pub struct First;

#[derive(Default, Clone, Component)]
pub struct Second;

#[derive(Default, Clone, Component)]
pub struct Third;

#[derive(Default, Clone, Component)]
pub struct Group;

#[allow(dead_code)]
fn main() {
    //this file is for common structs and functions used in the examples
}
