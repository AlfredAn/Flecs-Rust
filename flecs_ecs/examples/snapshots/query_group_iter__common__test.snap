---
source: flecs_ecs/examples/common.rs
expression: self.str
---
- All tables
- "group: \"::query_group_iter::Cell_0_0\" - Table [query_group_iter.Merchant, query_group_iter.Npc, (query_group_iter.WorldCell,query_group_iter.Cell_0_0)]"
- "group: \"::query_group_iter::Cell_0_1\" - Table [query_group_iter.Npc, query_group_iter.Beggar, (query_group_iter.WorldCell,query_group_iter.Cell_0_1)]"
- "group: \"::query_group_iter::Cell_0_1\" - Table [query_group_iter.Npc, query_group_iter.Soldier, (query_group_iter.WorldCell,query_group_iter.Cell_0_1)]"
- "group: \"::query_group_iter::Cell_1_0\" - Table [query_group_iter.Npc, query_group_iter.Mage, (query_group_iter.WorldCell,query_group_iter.Cell_1_0)]"
- "group: \"::query_group_iter::Cell_1_0\" - Table [query_group_iter.Npc, query_group_iter.Beggar, (query_group_iter.WorldCell,query_group_iter.Cell_1_0)]"
- "group: \"::query_group_iter::Cell_1_1\" - Table [query_group_iter.Npc, query_group_iter.Soldier, (query_group_iter.WorldCell,query_group_iter.Cell_1_1)]"
- "\n"
- "Tables for cell 1_0:"
- "group: \"::query_group_iter::Cell_1_0\" - Table [query_group_iter.Npc, query_group_iter.Mage, (query_group_iter.WorldCell,query_group_iter.Cell_1_0)]"
- "group: \"::query_group_iter::Cell_1_0\" - Table [query_group_iter.Npc, query_group_iter.Beggar, (query_group_iter.WorldCell,query_group_iter.Cell_1_0)]"
